### PR TITLE
added tableData and descriptions as permissions

### PR DESCRIPTION
**Pull request by rfontanarosa**
part of #1519

### CODE REVIEW

File: src/meta/user/authorizer.ts [hunk 1]
```diff
@@ -72,8 +72,14 @@ const canViewUsers = (props: { countryIso: CountryIso; assessment: Assessment; c
...
  * @returns boolean
  */
-const canEdit = (props: { countryIso: CountryIso; section: Section; country: Country; user: User }): boolean => {
-  const { section, user, countryIso, country } = props
+const canEdit = (props: {
+  countryIso: CountryIso
+  section: Section
+  country: Country
+  user: User
+  checkPermission?: 'tableData' | 'descriptions'
```
Comment (minotogna) [inline src/meta/user/authorizer.ts]: I agree, let's add the enum

Comment (rfontanarosa) [reply src/meta/user/authorizer.ts]: CollaboratorEditPropertyType

File: src/server/middleware/auth.ts [hunk 2]
```diff
@@ -146,6 +158,8 @@ const requireEditAssessmentFile = async (req: Request, _res: Response, next: Nex
 
 export const AuthMiddleware = {
   requireEdit,
```
Comment (minotogna) [inline src/server/middleware/auth.ts]: is it true, requireEdit now it's not used anymore from outside ?

Comment (rfontanarosa) [reply src/server/middleware/auth.ts]: it's used here:

-     express.post(ApiEndPoint.Area.country(), AuthMiddleware.requireEdit, postCountry)
-     express.post(ApiEndPoint.CycleData.OriginalDataPoint.one(), AuthMiddleware.requireEdit, createOriginalDataPoint)
- express.delete(ApiEndPoint.CycleData.OriginalDataPoint.one(), AuthMiddleware.requireEdit, deleteOriginalDataPoint)
- express.put(ApiEndPoint.CycleData.OriginalDataPoint.one(), AuthMiddleware.requireEdit, updateOriginalDataPoint)

Comment (minotogna) [reply src/server/middleware/auth.ts]: can u use requireEditTableData and make requireEdit internal to AuthMiddleware.requireEdit?

**Review by sorja — APPROVED** [review]

File: src/server/middleware/auth.ts (hunk 2 above)
Comment (rfontanarosa) [reply src/server/middleware/auth.ts]: done

**Review by minotogna — APPROVED** [review]

**Pull request merged**